src/main/java/org/kohsuke/github/GHRateLimit.java

```diff
@@ -308,31 +306,37 @@ private UnknownLimitRecord() {
          * @param resetEpochSeconds
          *            the reset epoch seconds
          */
-        @JsonCreator
         public Record(@JsonProperty(value = "limit", required = true) int limit,
                 @JsonProperty(value = "remaining", required = true) int remaining,
                 @JsonProperty(value = "reset", required = true) long resetEpochSeconds) {
             this(limit, remaining, resetEpochSeconds, null);
         }
 
         /**
-         * Instantiates a new Record.
+         * Instantiates a new Record. Called by Jackson data binding or during header parsing.
          *
          * @param limit
          *            the limit
          * @param remaining
          *            the remaining
          * @param resetEpochSeconds
          *            the reset epoch seconds
-         * @param updatedAt
-         *            the updated at
+         * @param responseInfo
+         *            the response info
          */
-        @SuppressFBWarnings(value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD", justification = "Deprecated")
-        public Record(int limit, int remaining, long resetEpochSeconds, @CheckForNull String updatedAt) {
+        @JsonCreator
+        Record(@JsonProperty(value = "limit", required = true) int limit,
+                @JsonProperty(value = "remaining", required = true) int remaining,
+                @JsonProperty(value = "reset", required = true) long resetEpochSeconds,
+                @JacksonInject @CheckForNull GitHubResponse.ResponseInfo responseInfo) {
```

Comment:
Also, since we now have the the `responseInfo` injected during binding, we can make `Record` fully immutable.  We don't have "recalculate" after creation.